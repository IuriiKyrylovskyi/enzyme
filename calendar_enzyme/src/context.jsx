import React, { useState, useContext } from "react";
import moment from "moment";

const AppContext = React.createContext();

const AppProvider = ({ children }) => {
  const [isOpen, setIsOpen] = useState(false);
  const [isEvent, setEvent] = useState(false);

  const [dateInput, setDateInput] = useState("");
  const [startTimeInput, setStartTimeInput] = useState("");
  const [endTimeInput, setEndTimeInput] = useState("");

  const onCloseModal = () => {
    setIsOpen(false);
  };

  const onOpenModal = (e) => {
    if (isEvent || e.target.closest(".event")) {
      return; // setEvent(false);
    }
    if (e.target.className === "calendar__time-slot") {
      // console.log(onOpenModalStartTime());
      setDateInput("");
      setStartTimeInput("");
      setEndTimeInput("");

      setEvent(false);
      setIsOpen(true);
      console.log(e);
      console.log(e.nativeEvent);
      return;
    }

    if (!e.target.closest("event")) {
      console.log(e.target.className);
      setDateInput(moment().format("YYYY-MM-DD"));
      setStartTimeInput(moment().format("HH:mm"));
      setEndTimeInput(moment().format("HH:mm"));

      setEvent(false);
      setIsOpen(true);
    }
  };

  // function onOpenModalStartTime(start) {
  //   setStartTimeInput(start);
  // }

  return (
    <AppContext.Provider
      value={{
        isOpen,
        isEvent,
        onCloseModal,
        onOpenModal,

        dateInput,
        startTimeInput,
        endTimeInput,
        // onOpenModalStartTime,
      }}
    >
      {children}
    </AppContext.Provider>
  );
};

export const useGlobalContext = () => {
  return useContext(AppContext);
};

export { AppProvider };
