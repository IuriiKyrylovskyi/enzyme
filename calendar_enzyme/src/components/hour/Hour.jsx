import React from "react";
import { useGlobalContext } from "../../context";
import TimeLine from "../timeLine/TimeLine";
import Event from "../event/Event";
// import Modal from "../modal/Modal";

// import { deleteEvent } from '../../gateway/gateway';
// import { formatMins } from "../../../src/utils/dateUtils.js";

const Hour = ({ weekStartDate, dataDay, dataHour, hourEvents, fetchEvents, deleteEvent }) => {
  const currentDate = weekStartDate.getFullYear() === new Date().getFullYear() && weekStartDate.getMonth() === new Date().getMonth() && dataDay === new Date().getDate() && dataHour === new Date().getHours();

  const { isOpen, onOpenModal, onOpenModalStartTime } = useGlobalContext();

  // const handleHour = () => {
  //   console.log('hour', dataHour);
  // console.log(onOpenModalStartTime(dataHour));
  // onOpenModalStartTime(dataHour);
  // };

  return (
    <div
      className="calendar__time-slot"
      data-time={dataHour + 1}
      onClick={onOpenModal}
      // onTarget={() => console.log(dataDay)}
      //
    >
      {/* if no events in the current hour nothing will render here */}
      {hourEvents.map(({ id, date, startTime, endTime, title }) => {
        const dateFrom = new Date(date + " " + startTime);
        const dateTo = new Date(date + " " + endTime);

        // console.log(id, date, startTime, endTime, title);
        // console.log((dateTo - dateFrom) / 1000 / 60 / 60);
        // const eventDuration = (dateTo - dateFrom) / 1000 / 60 / 60;

        return (
          //eventDuration <= 6 &&
          <Event
            key={id}
            id={id}
            //calculating event height = duration of event in minutes
            height={(dateTo.getTime() - dateFrom.getTime()) / (1000 * 60)}
            marginTop={dateFrom.getMinutes()}
            time={`${startTime} - ${endTime}`}
            title={title}
            fetchEvents={fetchEvents}
            handleDelete={deleteEvent}
            // handleDelete={onDeleteEvent}
          />
        );
      })}
      {currentDate && <TimeLine weekStartDate={weekStartDate} />}
      {/* {isOpen && <Modal fetchEvents={fetchEvents} />} */}
      {/* <TimeLine weekStartDate={weekStartDate} /> */}
    </div>
  );
};

export default Hour;
