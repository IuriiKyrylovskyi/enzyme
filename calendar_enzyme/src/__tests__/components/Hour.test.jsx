import React from "react";
import { mount } from "enzyme";
import { AppProvider } from "../../context";
import Hour from "../../components/hour/Hour";

describe("<Hour />", () => {
  const props = {
    weekStartDate: new Date("2019-04-22T10:20:30Z"),
    weekDates: [new Date("2019-04-22T10:20:30Z"), new Date("2019-04-23T10:20:30Z"), new Date("2019-04-24T10:20:30Z"), new Date("2019-04-25T10:20:30Z"), new Date("2019-04-26T10:20:30Z"), new Date("2019-04-27T10:20:30Z"), new Date("2019-04-28T10:20:30Z")],
    hourEvents: [{ createdAt: "2019-04-23T10:20:30Z", date: "2019-04-23", description: "", endTime: "10:00", id: "30", name: "Kenneth Wiegand", startTime: "04:00", title: "to do something" }],
    fetchEvents: jest.fn(() => Promise.resolve()),
    deleteEvent: jest.fn(),
    dataDay: 23,
    dataHour: 10,
  };

  const wrappedComponent = mount(
    <AppProvider>
      <Hour {...props} />
    </AppProvider>
  );

  it("should display component", () => {
    expect(wrappedComponent).toMatchSnapshot();
  });

  it("should not open modal on click", () => {
    const onOpenModal = jest.fn();
    wrappedComponent.find(".calendar__time-slot div").at(0).prop("onClick");

    expect(onOpenModal).not.toBeCalled();
  });
});

//---------------------------------------------

describe("Hour component", () => {
  const props = {
    weekStartDate: new Date("2019-04-22T10:20:30Z"),
    weekDates: [new Date("2019-04-22T10:20:30Z"), new Date("2019-04-23T10:20:30Z"), new Date("2019-04-24T10:20:30Z"), new Date("2019-04-25T10:20:30Z"), new Date("2019-04-26T10:20:30Z"), new Date("2019-04-27T10:20:30Z"), new Date("2019-04-28T10:20:30Z")],
    hourEvents: [],
    fetchEvents: jest.fn(() => Promise.resolve()),
    deleteEvent: jest.fn(),
    dataDay: 23,
    dataHour: 10,
  };

  const setUp = () =>
    mount(
      <AppProvider>
        <Hour {...props} />
      </AppProvider>
    );

  let component;
  let instance;

  beforeEach(() => {
    component = setUp();
    instance = component.instance();
  });

  it("should render Hour component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should open modal on click", () => {
    const btn = component.find(".calendar__time-slot div").at(0);
    btn.simulate("click");

    expect(component).toMatchSnapshot();
  });
});
