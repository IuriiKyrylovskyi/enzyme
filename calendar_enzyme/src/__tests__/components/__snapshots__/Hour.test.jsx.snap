// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hour /> should display component 1`] = `
<AppProvider>
  <Hour
    dataDay={23}
    dataHour={10}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    hourEvents={
      Array [
        Object {
          "createdAt": "2019-04-23T10:20:30Z",
          "date": "2019-04-23",
          "description": "",
          "endTime": "10:00",
          "id": "30",
          "name": "Kenneth Wiegand",
          "startTime": "04:00",
          "title": "to do something",
        },
      ]
    }
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  >
    <div
      className="calendar__time-slot"
      data-time={11}
    >
      <Event
        fetchEvents={[MockFunction]}
        handleDelete={[MockFunction]}
        height={360}
        id="30"
        key="30"
        marginTop={0}
        startTime={1555981200000}
        time="04:00 - 10:00"
        title="to do something"
      >
        <div
          className="event"
          onClick={[Function]}
          style={
            Object {
              "height": 360,
              "marginTop": 0,
            }
          }
        >
          <div
            className="event__title"
          >
            to do something
          </div>
          <div
            className="event__time"
          >
            04:00 - 10:00
          </div>
        </div>
      </Event>
    </div>
  </Hour>
</AppProvider>
`;

exports[`<Hour /> should open modal on click 1`] = `
<AppProvider>
  <Hour
    dataDay={23}
    dataHour={10}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    hourEvents={
      Array [
        Object {
          "createdAt": "2019-04-23T10:20:30Z",
          "date": "2019-04-23",
          "description": "",
          "endTime": "10:00",
          "id": "30",
          "name": "Kenneth Wiegand",
          "startTime": "04:00",
          "title": "to do something",
        },
      ]
    }
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  >
    <div
      className="calendar__time-slot"
      data-time={11}
    >
      <Event
        fetchEvents={[MockFunction]}
        handleDelete={[MockFunction]}
        height={360}
        id="30"
        key="30"
        marginTop={0}
        startTime={1555981200000}
        time="04:00 - 10:00"
        title="to do something"
      >
        <div
          className="event"
          onClick={[Function]}
          style={
            Object {
              "height": 360,
              "marginTop": 0,
            }
          }
        >
          <div
            className="event__title"
          >
            to do something
          </div>
          <div
            className="event__time"
          >
            04:00 - 10:00
          </div>
        </div>
      </Event>
    </div>
  </Hour>
</AppProvider>
`;

exports[`Hour component should open modal on click 1`] = `
<AppProvider>
  <Hour
    dataDay={23}
    dataHour={10}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    hourEvents={Array []}
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  >
    <div
      className="calendar__time-slot"
      data-time={11}
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </Hour>
</AppProvider>
`;

exports[`Hour component should render Hour component 1`] = `
<AppProvider>
  <Hour
    dataDay={23}
    dataHour={10}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    hourEvents={Array []}
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  >
    <div
      className="calendar__time-slot"
      data-time={11}
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </Hour>
</AppProvider>
`;
